### DISCUSSION TITLE
Publishing on premises

### DISCUSSION THREAD

**Discussion by pvi-github** [webstudio-is/webstudio-community#144] [Q&A]
Hello Webstudio community.

The designer is really interesting but currently it can publish to "vanilla", "vercel.com" and "netlify.com"

How can I publish on my company nodejs servers ?

The code is available through the webstudio tool and command line but the assets are referenced with urls to webstudio.is domain :-/

(And by the way all these assets like pictures are referenced remotely in the html code produced by npm run dev and unavailable due to security policy. Even adding https to vite does not solve this problem due to browser security rules. It would be better to be able to run and build autonomous 
 apps with relative pathes to local assets)

**Comment by kof** [comment]
I think we looked into your case via discord and identified that you used absolute urls, basically you hardcoded webstudio.is urls into those images. We don't do that, all assets are synced locally and used locally when you are using the CLI
